src/main/scala/com/bazaarvoice/sswf/service/StepDecisionWorker.scala

```diff
@@ -179,14 +185,15 @@ class StepDecisionWorker[SSWFInput, StepEnum <: (Enum[StepEnum] with WorkflowSte
       if (finalStates.isEmpty) {
         finalStates.append(e)
       } else {
-        (finalStates.last.event.left.get, e.event.left.get) match {
-          case (lastDS: DefinedStep[StepEnum], eventDS: DefinedStep[StepEnum]) if lastDS.withoutResume == eventDS.withoutResume =>
+        val previousFinalState: StepEvent[StepEnum] = finalStates.last
+        (previousFinalState.event.left.get, e.event.left.get) match {
+          case (lastDS: DefinedStep[StepEnum], eventDS: DefinedStep[StepEnum]) if lastDS.withoutResume == eventDS.withoutResume && StepResult.deserialize(previousFinalState.result).isInProgress =>
```

Comment:
is withoutResult with == the same comparison as isSameStepAs? Seems like we should consolidate on the simpler expression.